pgrx-examples/decoder/src/lib.rs

```diff
@@ -0,0 +1,322 @@
+use pgrx::prelude::*;
+use serde::Serialize;
+use serde::ser::{Serializer, SerializeStruct};
+use std::alloc::{alloc, dealloc, Layout};
+
+::pgrx::pg_module_magic!();
+
+// An Action describe a change that occured on a table
+#[derive(Serialize)]
+struct Action {
+    typ: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    committed: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    rel: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    old: Option<Tuple>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    new: Option<Tuple>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    change_count: Option<i64>,
+}
```

Comment:
Can you add a unit test to this crate that attempts to serialize this struct, with all the Options as None, and then deserialize this struct from the resulting string?